CMakeLists.txt

```diff
@@ -212,29 +144,26 @@ else()
   message(FATAL_ERROR "Required target protobuf::libprotobuf-lite or protobuf::libprotobuf is not defined")
 endif()
 
-add_library(libprime_server INTERFACE IMPORTED)
+# libprime_server
+# see https://gitlab.kitware.com/cmake/cmake/-/issues/19467
+set(libprime_server_targets "")
 if(ENABLE_SERVICES)
-  pkg_check_modules(libprime_server REQUIRED libprime_server>=0.6.3)
-  # workaround for https://gitlab.kitware.com/cmake/cmake/issues/15804
-  find_library(libprime_server_LIBRARY
-    NAME ${libprime_server_LIBRARIES}
-    HINTS ${libprime_server_LIBRARY_DIRS})
-  set_target_properties(libprime_server PROPERTIES
-    INTERFACE_LINK_LIBRARIES "${libprime_server_LIBRARY}"
-    INTERFACE_INCLUDE_DIRECTORIES "${libprime_server_INCLUDE_DIRS}"
-    INTERFACE_COMPILE_DEFINITIONS HAVE_HTTP)
+  pkg_check_modules(libprime_server IMPORTED_TARGET libprime_server>=0.6.3)
+  if (libprime_server_FOUND)
+    target_compile_definitions(PkgConfig::libprime_server INTERFACE ENABLE_SERVICES)
+    set(libprime_server_targets PkgConfig::libprime_server)
+  else()
+    set(ENABLE_SERVICES OFF)
+    message(WARNING "prime_server not found, skipping ENABLE_SERVICES...")
```

Comment:
same point, i dont think its a good idea to just warn and move on, this is major functionality within the library being disabled. the immediate result of this will be 20 github issues about "i built valhalla and the docs say to run valhalla_service but i cant find it"